tls/s2n_x509_validator.c

```diff
@@ -319,7 +319,7 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
 
         struct s2n_blob asn1cert = {0};
         asn1cert.size = certificate_size;
-        asn1cert.data = s2n_stuffer_raw_read(&cert_chain_in_stuffer, certificate_size);
```

Comment:
looks like the next line is checking for NULL. seems like adding `notnull_check()` would alter the error code returned here.